src/views/Player.vue

```diff
@@ -267,28 +263,50 @@ export default defineComponent({
       if (profile.value && battleTag.value === profile.value.battleTag) return;
 
       await playerStore.loadProfile({ battleTag: battleTag.value, freshLogin: props.freshLogin });
-      await playerStore.loadGameModeStats({});
-      await playerStore.loadRaceStats();
-      await playerStore.loadPlayerStatsRaceVersusRaceOnMap(battleTag.value);
-      await playerStore.loadPlayerStatsHeroVersusRaceOnMap(battleTag.value);
-      await playerStore.loadPlayerGameLengths();
-      await playerStore.loadMatches(1);
+      await loadAllDataForSelectedSeason();
       await initMmrRpTimeline();
       window.scrollTo(0, 0);
     }
 
+    async function loadAllDataForSelectedSeason() {
+      await Promise.all([
+        playerStore.loadGameModeStats({}),
+        playerStore.loadRaceStats(),
+        playerStore.loadMatches(1),
+        playerStore.loadPlayerStatsRaceVersusRaceOnMap(battleTag.value),
+        playerStore.loadPlayerStatsHeroVersusRaceOnMap(battleTag.value),
+        playerStore.loadPlayerGameLengths(),
+      ]);
+    }
+
     async function initMmrRpTimeline() {
-      const raceStats = playerStore.raceStats;
+      // Make a lookup table for active game modes
+      const activeGameModes = rankingsStore.activeModes.reduce((acc, mode) => {
```

Comment:
Is it possible to use the helper function `activeGameModes` here?
https://github.com/w3champions/website/blob/master/src/mixins/GameModesMixin.ts#L24

Reply:
Fixed

Reply:
Yeah okay that doesn't work actually because calling `activeGameModes()` here throws `SyntaxError: Must be called at the top of a setup function`. We already have the data loaded in the rankingsStore so we can just fetch it directly, no?

Reply:
Yeah okay there's an issue, you're right, needs to call it since it happens in `Player.vue` but the data is normally only loaded in `PlayerStatisticTab.vue` (I've been refreshing that tab specifically so I didn't notice that it didn't load it in time

Reply:
It's complaining on this line https://github.com/w3champions/website/blob/52dd50e06a1f29da64597cc4789e2c548b297a8d/src/mixins/GameModesMixin.ts#L71 because `useI18n()` apparently needs to be called top-level. Seems like a bug with `getGameModes()` :grimacing:

Reply:
I'll take a look in a few hours.

Reply:
Well the bug is fixed by calling `rankingsStore.retrieveRankings()` in `Player.vue`, but I don't think we can use the mixin because of how it wants to use i18n for whatever reason. I don't understand that error. We only need the IDs here so this is enough.

Reply:
That's annoying. That's a bug I introduced when I moved all the components to the Composition API to prepare for the transition from Vue 2 to Vue 3. But I haven't continued the migration yet, because there's so much breakage related to Vuetify.